migrations/state_test.go

```
@@ -0,0 +1,264 @@
+		diff = -diff
+	}
+	if diff > time.Second {
+		t.Errorf("expected LastRun within 1 second, got difference of %v", diff)
+	}
+}
+
+func TestStateManager_Load_InvalidJSON(t *testing.T) {
+	tmpDir := t.TempDir()
+	statePath := filepath.Join(tmpDir, "migrations.json")
+
+	err := os.WriteFile(statePath, []byte("invalid json"), 0644)
```

golangci-lint (2.5.0)

[medium] G306: Expect WriteFile permissions to be 0600 or less

(gosec)